index.js

```diff
@@ -732,21 +862,21 @@ Peer.prototype._onChannelClose = function () {
   self.destroy()
 }
 
-Peer.prototype._onAddStream = function (event) {
-  var self = this
-  if (self.destroyed) return
-  self._debug('on add stream')
-  self.emit('stream', event.stream)
-}
-
 Peer.prototype._onTrack = function (event) {
   var self = this
   if (self.destroyed) return
+
+  self._remoteTracks.push(event.track)
   self._debug('on track')
-  var id = event.streams[0].id
-  if (self._previousStreams.indexOf(id) !== -1) return // Only fire one 'stream' event, even though there may be multiple tracks per stream
-  self._previousStreams.push(id)
-  self.emit('stream', event.streams[0])
+  self.emit('track', event.track)
+
+  if (self._remoteStreams.some(function (stream) {
+    return stream.id === event.streams[0].id // Only fire one 'stream' event, even though there may be multiple tracks per stream
+  })) return
+  self._remoteStreams.push(event.streams[0])
```

Comment:
Spec says one track might be a part of more than 1 stream. Why do we only care about one of them?

Reply:
The comment there is referring to only firing one stream event for multiple track events that reference the same stream. We did that before, I've just refactored it.

Reply:
The spec says there might be `event.streams[1]`, otherwise it wouldn't be called that way. I'm just curious what would happen if we have multiple tracks, some of which are used in multiple streams. I don't think it will happen with high probability, but it is still a valid edge case.